### PR TITLE
Add exception types to except clauses.

### PR DESCRIPTION
**Pull request by skschneider**

### CODE REVIEW

**Comment by noahl** [comment]
The object storage benchmark part LGTM

File: perfkitbenchmarker/providers/__init__.py [hunk 1]
```diff
@@ -67,6 +67,6 @@ def LoadProvider(provider_name, ignore_package_requirements=True):
                                               __name__ + '.' + provider_name)]
     if not modules:
       raise ImportError('No modules found for provider.')
-  except:
+  except ImportError:
```
Comment (ehankland) [inline perfkitbenchmarker/providers/__init__.py]: I think this was broad on purpose. Not all errors that you encounter while loading modules are import errors - if the python file exists, but has invalid syntax for example, I don't think that will raise an ImportError. In that case, I think it's still useful to have a message that tells you that the provider failed to load.

File: perfkitbenchmarker/traces/dstat.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def _StopOnVm(self, vm):
     vm.RemoteCommand(cmd)
     try:
       vm.PullFile(self.output_directory, file_name)
-    except:
+    except Exception:
```
Comment (ehankland) [inline perfkitbenchmarker/traces/dstat.py]: I think PullFile calls RemoteCopy which raises errors.VirtualMachine.RemoteCommandError on failure.

Comment (skschneider) [inline perfkitbenchmarker/traces/dstat.py]: Changed to Exception. I'm just trying to cut down on places where things like KeyboardInterrupt or SystemExit can be involuntarily swallowed.

File: perfkitbenchmarker/providers/__init__.py (hunk 1 above)
Comment (skschneider) [inline perfkitbenchmarker/providers/__init__.py]: Changed to Exception.

File: perfkitbenchmarker/traces/dstat.py (hunk 2 above)
Comment (skschneider) [inline perfkitbenchmarker/traces/dstat.py]: Discussed offline. As a separate issue, we should address errors.VirtualMachine.RemoteExceptionError, which can also be raised in some situations.

**Comment by ehankland** [comment]
LGTM

**Pull request merged**